actors/miner/src/lib.rs

```diff
@@ -1950,6 +1958,283 @@ impl Actor {
+                "unsupported seal proof type {}",
+                i64::from(params.seal_proof_type)
+            ));
+        }
+
+        if params.aggregate_proof_type != RegisteredAggregateProof::SnarkPackV2 {
+            return Err(actor_error!(illegal_argument, "aggregate proof type must be SnarkPackV2"));
+        }
+
+        validate_seal_aggregate_proof(
+            &params.aggregate_proof,
+            params.sectors.len() as u64,
+            policy,
+            true,
+        )?;
+
+        if params.sectors.iter().any(|sector| sector.sealer_id != miner_id) {
+            return Err(actor_error!(
+                illegal_argument,
+                "sealer must be the same as the receiver actor for all sectors"
+            ));
+        }
+
+        if params.sectors.iter().any(|sector| sector.sector_number != sector.sealing_number) {
+            return Err(actor_error!(
+                illegal_argument,
+                "sealing number must be same as sector number for all sectors"
+            ));
+        }
+
+        let curr_epoch = rt.curr_epoch();
+        let challenge_earliest = curr_epoch - rt.policy().max_prove_commit_ni_randomness_lookback;
+        let unsealed_cid = CompactCommD::empty().get_cid(params.seal_proof_type).unwrap();
+        let entropy = serialize(&rt.message().receiver(), "address for get verify info")?;
+
+        for sector in params.sectors.iter() {
+            if !is_sealed_sector(&sector.sealed_cid) {
+                return Err(actor_error!(illegal_argument, "sealed CID had wrong prefix"));
+            }
+
+            if sector.seal_rand_epoch >= curr_epoch {
+                return Err(actor_error!(
+                    illegal_argument,
+                    "seal challenge epoch {} must be before now {}",
+                    sector.seal_rand_epoch,
+                    curr_epoch
+                ));
+            }
+
+            if sector.seal_rand_epoch < challenge_earliest {
+                return Err(actor_error!(
+                    illegal_argument,
+                    "seal challenge epoch {} too old, must be after {}",
+                    sector.seal_rand_epoch,
+                    challenge_earliest
+                ));
+            }
+        }
+
+        let state: State = rt.state()?;
+        let store = rt.store();
+        let info = get_miner_info(rt.store(), &state)?;
+        let activation_epoch = rt.curr_epoch();
+
+        if params.proving_deadline >= policy.wpost_period_deadlines {
+            return Err(actor_error!(
+                illegal_state,
+                "proving deadline index {} invalid",
+                params.proving_deadline
+            ));
+        }
+
+        if !deadline_is_mutable(
+            policy,
+            state.current_proving_period_start(policy, curr_epoch),
+            params.proving_deadline,
+            curr_epoch,
+        ) {
+            return Err(actor_error!(
+                illegal_argument,
```

Comment:
Even though sector termination uses illegal argument here, I think forbidden would be better. It's well-formed, just not allowed right now.